### DISCUSSION TITLE
How can reactive programming be enabled for key-cache?

### DISCUSSION THREAD

**Discussion by dwang2** [MarcGiffing/bucket4j-spring-boot-starter#286] [General]
I have developed a reactive security service to retrieve the current user's name using the code snippet provided below. How can I formulate a key-cache expression to asynchronously fetch the username

> @Service
public class SecurityService {

	public Mono<String> username() {
		return ReactiveSecurityContextHolder.getContext()
				.map(securityContext -> {
					Authentication authentication = securityContext.getAuthentication();
					return authentication != null ? authentication.getName() : null;
				});
	}
	
}

**Comment by MarcGiffing** [comment]
This essential requirements is not supported yet. See #289.
